base_layer/core/src/validation/transaction_validators.rs

```diff
@@ -125,6 +126,20 @@ fn verify_not_stxos<B: BlockchainBackend>(tx: &Transaction, db: &B) -> Result<()
     Ok(())
 }
 
+/// This function checks the at the tx contains no duplicated inputs or outputs.
+fn verify_no_duplicated_inputs_outputs(tx: &Transaction) -> Result<(), ValidationError> {
+    let mut mut_copy = tx.clone();
+    if !mut_copy.body.check_duplicated_inputs() {
```

Comment:
Personal opinion: I think these should be named like `contains_duplicated_inputs` and swap around the returning of true and false. 

Then this line becomes `if mut_copy.body.contains_duplicated_inputs()` which I think is more intuitive.